### ISSUE TITLE
no-else-return fixed to incorrect code

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#11069]
**Tell us about your environment**

* **ESLint Version:** 5.8.0/master
* **Node Version:** n/a
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  rules: {"no-else-return": 2}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function foo(){
    let bar = 0;
    if(cond) {
        return bar;
    } else {
        let bar = 1;
        return bar;
    }
}
```

```bash

```

**What did you expect to happen?**
not fix to incorrect code.

**What actually happened? Please include the actual, raw output from ESLint.**

fixed code:
```js
function foo(){
    let bar = 0;
    if(cond) {
        return bar;
    } 
        let bar = 1;
        return bar;
    
}
```

```bash
7:13  error  Parsing error: Identifier 'bar' has already been declared
```

**Are you willing to submit a pull request to fix this bug?**
yes, but I don't have time to do it in the following days. please feel free to take it if someone else have interests.

**Comment by SerhiiBilyk** [comment]
I can take it

**Comment by Aladdin-ADD** [comment]
@SerhiiBilyk thanks for reaching out!

the related code is the fixer function:  https://github.com/eslint/eslint/blob/5da378ac922d732ca1765f08edee0face1b1b924/lib/rules/no-else-return.js#L63-L64

it should return `null`, if not safe to do the fix.

**Comment by SerhiiBilyk** [comment]
@Aladdin-ADD thank you for the answer.
Do I need to remove last [let] keyword in case if same variable exist in current scope?

**Comment by Aladdin-ADD** [comment]
I think just as-is would be fine. `return null;` in fixer function means not to fix it. (I should be more clear 😄

**Comment by SerhiiBilyk** [comment]
Aladdin-ADD , ok I have understand, thank you again!

**Comment by SerhiiBilyk** [comment]
Aladdin-ADD , ok I have understand, thank you again!

**Comment by SerhiiBilyk** [comment]
@Aladdin-ADD can you check the algorithm.
I'm cheking if there block scope local variable declaration inside [else] statement.
If [true] return null

Is this correct?

```js
const isScopeDeclaration = (token) => {
  const keyword = token.type === 'Keyword';
  const scopeDeclaration = token.value === 'let' || token.value === 'const';
  return keyword && scopeDeclaration;
};
const isDeclarationInside = sourceCode.getTokensBetween(startToken, endToken).findIndex(isScopeDeclaration) > -1;
```

**Comment by Aladdin-ADD** [comment]
SGTM! Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have some more questions!

and thanks for doing this!

**Comment by platinumazure** [comment]
Reopening since we have an open PR.
